imod/mf6/uzf.py

```diff
@@ -341,16 +352,19 @@ def _check_options(
 
     def _create_uzf_numbers(self, landflag):
         """Create unique UZF ID's. Inactive cells equal 0"""
-        return np.cumsum(np.ravel(landflag)).reshape(landflag.shape) * landflag
+        active_nodes = xr.ones_like(landflag).where(landflag.notnull(), other =0.0)
+        return np.nancumsum(active_nodes).reshape(landflag.shape) * active_nodes
 
     def _determine_landflag(self, kv_sat):
-        return (np.isfinite(kv_sat)).astype(np.int32)
+        """ returns the landflag for uzf-model. Landflag == 1 for top active UZF-nodes """
+        land_nodes = xr.ones_like(kv_sat,dtype = np.int32).where(kv_sat.layer == get_upper_active_layer_number(kv_sat),other = 0)
```

Comment:
[This can be shortened further with this utility function](https://deltares.github.io/imod-python/api/generated/prepare/imod.prepare.get_upper_active_grid_cells.html)

```suggestion
        land_nodes = get_upper_active_grid_cells(kv_sat).astype(np.int32)
```

Reply:
done